lambdas/db-migration/src/migrations/20201014105024_create_rules_table.ts

```diff
@@ -0,0 +1,66 @@
+import * as Knex from 'knex';
+
+export const up = async (knex: Knex): Promise<void> =>
+  knex.schema.createTable('rules', (table) => {
+    table
+      .increments('cumulusId')
+      .primary();
+    table
+      .text('name')
+      .comment('Rule name')
+      .notNullable();
+    table
+      .text('workflow')
+      .comment('Workflow name to invoke for this rule')
+      .notNullable();
+    table
+      .integer('collectionCumulusId')
+      .references('cumulusId')
+      .inTable('collections')
+      .notNullable();
+    table
+      .integer('providerCumulusId')
+      .references('cumulusId')
+      .inTable('providers')
+      .notNullable();
+    table
+      .enum('type', ['onetime', 'scheduled', 'sns', 'kinesis', 'sqs'])
+      .comment(
+        'Specifies how workflows are invoked for this rule'
+      )
+      .notNullable();
+    table
+      .boolean('enabled')
+      .comment('Whether rule is active or not')
+      .notNullable();
+    table
+      .text('value')
+      .comment(`
+        Value is multi-use. For a kinesis rule this is the target stream arn,
+        for a scheduled event it is the schedule pattern (e.g. cron), for a one-time rule.
+      `);
+    table
+      .text('arn')
+      .comment('For kinesis rules: ARN of event source mapping between Kinesis stream and message consumer Lambda');
+    table
+      .text('logEventArn')
+      .comment('For kinesis rules: ARN of event source mapping between Kinesis stream and inbound event logger Lambda');
+    table
+      .jsonb('payload')
+      .comment('Optional input payload to use for onetime and scheduled workflows');
+    table
+      .jsonb('meta')
+      .comment('Optional metadata for the rule. Contents will be automatically added to $.meta on invoked workflows.');
+    table
+      .jsonb('tags')
+      .comment('Optional tags for the rule');
+    table
+      .text('queueUrl')
+      .comment('Optional SQS queue URL used to schedule executions for this rule');
+    table
+      .timestamps(false, true);
```

Comment:
Can you explain the preference for datetime over timestamp for my understanding?

Reply:
For Knex, I'm not sure if it makes a difference:

http://knexjs.org/#Schema-datetime
http://knexjs.org/#Schema-timestamp

However, using `timestamps` is nice because it creates the `created_at` and `updated_at` columns automatically:

http://knexjs.org/#Schema-timestamps

PostgreSQL docs on date/time column types:

https://www.postgresql.org/docs/11/datatype-datetime.html